packages/contracts-bedrock/src/L2/CrossL2Inbox.sol

```diff
@@ -133,12 +139,40 @@ contract CrossL2Inbox is ISemver, TransientReentrancyAware {
     /// @param _id      Identifier of the message.
     /// @param _msgHash Hash of the message payload to call target with.
     function validateMessage(Identifier calldata _id, bytes32 _msgHash) external {
-        // We need to know if this is being called on a depositTx
-        if (IL1BlockInterop(Predeploys.L1_BLOCK_ATTRIBUTES).isDeposit()) revert NoExecutingDeposits();
+        bytes32 checksum = calculateChecksum(_id, _msgHash);
+
+        (bool _isSlotWarm,) = _isWarm(checksum);
+
+        if (!_isSlotWarm) revert NotWarm();
 
         emit ExecutingMessage(_msgHash, _id);
     }
 
+    function _isWarm(bytes32 _slot) internal view returns (bool isWarm, uint256 result) {
+        assembly {
+            let startGas := gas()
+            // storing and returning the result so that the compiler doesn't optimize out the sload, this adds cost to
+            // the read
+            result := sload(_slot)
+            let endGas := gas()
+            isWarm := iszero(gt(sub(startGas, endGas), WARM_READ_COST))
+        }
+    }
+
+    bytes32 constant TYPE_3_MASK = 0x03ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff;
+
+    function calculateChecksum(Identifier memory _id, bytes32 _msgHash) public pure returns (bytes32) {
+        bytes32 logHash = keccak256(abi.encodePacked(_id.origin, _msgHash));
+
+        bytes memory idPacked = abi.encodePacked(_id.logIndex, _id.timestamp, _id.blockNumber);
+
```

Comment:
Current spec defines `idPacked` as being left-padded to 32 bytes.

The only reason why there is a packed 20+32 byte input in `logHash` computation is because of a legacy DB choice. Which maybe we should just break, to introduce consistency.

The thought is that the rest should all just be pairs of 32 bytes, hashing 64 bytes together each time.